src/main/scala/org/broadinstitute/dsde/firecloud/model/ElasticSearch.scala

```diff
@@ -9,11 +9,19 @@ import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
 
 case class AttributeDefinition(properties: Map[String, AttributeDetail])
 
-case class AttributeDetail(`type`: String, items: Option[AttributeDetail] = None)
+case class AttributeDetail(`type`: String, items: Option[AttributeDetail] = None, aggregate: Option[Boolean] = None)
 
-case class ESDatasetProperty(properties: Map[String, ESDetail])
 
-case class ESDetail(`type`: String)
+trait ESPropertyFields
+case class ESDatasetProperty(properties: Map[String, ESPropertyFields])
+case class ESType(`type`: String) extends ESPropertyFields
+case class ESAggregatableType(`type`: String, fields:ESRaw) extends ESPropertyFields {
+  def this(str:String) = this(str, new ESRaw(str))
+}
+case class ESRaw(raw: ESAggregateProperties) {
+  def this(str: String) = this(ESAggregateProperties(str, "not_analyzed"))
+}
+case class ESAggregateProperties(`type`: String, index:String)
```

Comment:
There is actually a reason - not sure if you have run across this before. If I create a companion object for a type that needs a json format, the standard formater no longer references the class. instead it references the object. (at least this is what i think is happening). I can change the name of the object to solve this problem, but then this is no longer a companion object... but maybe that is still preferable.

this is the code addition just to demonstrate the error:
object ESRaw{}
the error happens whether this is declared before or after the class
here is the compiler error:
app_1    | [error] /app/src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelJsonProtocol.scala:351: type mismatch;
app_1    | [error]  found   : org.broadinstitute.dsde.firecloud.model.ESRaw.type
app_1    | [error]  required: ? => ?
app_1    | [error]  Note: implicit value ESRawFormat is not applicable here because it comes after the application point and it lacks an explicit result type
app_1    | [error]   implicit val ESRawFormat = jsonFormat1(ESRaw)